gpflow/utilities/multipledispatch.py

```diff
@@ -0,0 +1,38 @@
+from multipledispatch import Dispatcher as GeneratorDispatcher
+from multipledispatch.dispatcher import variadic_signature_matches
+from multipledispatch.variadic import isvariadic
+
+
+class Dispatcher(GeneratorDispatcher):
+    """
+    Multipledispatch's Dispatcher uses a generator to yield the 
+    desired function implmentation, which is problematic as TF
+    autograph is not able to compile code that passes through generators.
+
+    This class overwrites the problematic method in the original
+    Dispatcher and makes soley use of simple for-loops, which are
+    compilable by AutoGraph.
+    """
+
+    def dispatch(self, *types):
+        """
+        Returns matching function for `types`, if not existins returns None.
+        """
+        if types in self.funcs:
+            return self.funcs[types]
+
+        return self.get_first_occurrence(*types)
+
+    def get_first_occurrence(self, *types):
```

Comment:
Is that a copy from dispatcher? Can you explain what does it do?

Reply:
It is based on `dispatch_iter` from the Dispatcher but instead of returning a iterator/generator it returns the first element as is.